support-workers/src/typescript/services/salesforce.ts

```diff
@@ -140,68 +165,144 @@ export class SalesforceService {
 		giftRecipient: GiftRecipient | null,
 		user: User,
 	): Promise<SuccessfulUpsertResponse> | null {
-		if (giftRecipient?.firstName && giftRecipient.lastName) {
-			const giftRecipientContact: DeliveryContactRecordRequest = {
-				AccountId: contactRecord.AccountId,
-				Email: giftRecipient.email,
-				Salutation: giftRecipient.title,
-				FirstName: giftRecipient.firstName,
-				LastName: giftRecipient.lastName,
-				MailingStreet: user.deliveryAddress
-					? getAddressLine(user.deliveryAddress)
-					: null,
-				MailingCity: user.deliveryAddress?.city ?? null,
-				MailingState: user.deliveryAddress?.state ?? null,
-				MailingPostalCode: user.deliveryAddress?.postCode ?? null,
-				MailingCountry: user.deliveryAddress
-					? getCountryNameByIsoCode(user.deliveryAddress.country)
-					: null,
-				RecordTypeId: '01220000000VB50AAG',
-			};
+		if (giftRecipient && validGiftRecipientFields(giftRecipient)) {
+			const giftRecipientContact = createGiftRecipientContactRecordRequest(
+				contactRecord,
+				giftRecipient,
+				user,
+			);
 			return this.upsert(giftRecipientContact);
 		}
 		return null;
 	}
 }
 
-export const createContactRecordRequest = (
+export const validGiftRecipientFields = (
+	giftRecipient: GiftRecipient,
+): boolean => {
+	return !!giftRecipient.firstName && !!giftRecipient.lastName;
+};
+
+export const createBuyerRecordRequest = (
+	user: User,
+	buyerType: 'Print' | 'GiftBuyer' | 'DigitalOnly',
+):
+	| PrintContactRecordRequest
+	| GiftBuyerContactRecordRequest
+	| DigitalOnlyContactRecordRequest => {
+	switch (buyerType) {
+		case 'Print':
+			return createPrintContactRecordRequest(user);
+		case 'GiftBuyer':
+			return createGiftBuyerContactRecordRequest(user);
+		case 'DigitalOnly':
+			return createDigitalOnlyContactRecordRequest(user);
+	}
+};
+
+const getBuyerType = (
+	user: User,
+	hasGiftRecipient: boolean,
+): 'Print' | 'GiftBuyer' | 'DigitalOnly' => {
```

Comment:
This sum type is defined in a couple of places. Perhaps it should be given a name?

```ts
type BuyerType = 'Print' | 'GiftBuyer' | 'DigitalOnly';
```